src/libs/actions/Session/index.js

```diff
@@ -23,6 +23,31 @@ Onyx.connect({
     callback: val => credentials = val || {},
 });
 
+/**
+ * Manage push notification subscriptions on sign-in/sign-out.
+ *
+ * On Android, AuthScreens unmounts when the app is closed with the back button so we manage the
+ * push subscription when the session changes here.
+ */
+let previousAccountID;
+Onyx.connect({
+    key: ONYXKEYS.SESSION,
+    callback: (session) => {
+        const accountID = lodashGet(session, 'accountID');
+
+        if (accountID && previousAccountID !== accountID) {
```

Comment:
I wonder if we early return if `previousAccountID !== accountID` is false, then use an if/else condition for the register/deregister logic.

Reply:
```
if (previousAccountID == accountID) {
    return; 
}

if (accountID) {
    PushNotification.register(accountID);
} else {
    PushNotification.deregister();
    PushNotification.clearNotifications();
}
```

Reply:
+1 I like this suggestion